cardano-node/src/Cardano/Node/Run.hs

```diff
@@ -141,6 +142,17 @@ runNode loggingLayer npm@NodeCLI{protocolFiles} = do
       LiveView   -> do
 #ifdef UNIX
         let c = llConfiguration loggingLayer
+        -- check required tracers are turned on
+        let reqtrs = [("TraceBlockFetchDecisions",traceBlockFetchDecisions), ("TraceChainDb",traceChainDB), ("TraceForge",traceForge), ("TraceMempool",traceMempool)]
+            trsinactive = filter (\(_,f) -> not $ traceEnabled (ncTraceConfig nc) f) reqtrs
+        when (length trsinactive > 0) $ do
```

Comment:
Probably does not matter in this case but `length` forces the evaluation of the list in order to calculate the length.

A better solution would be:
```
     unless (List.null trsinactive) $ do
```
because that just inspects the head of the list to determine if it is empty or not.

Reply:
thanks, much better! :+1: